tools/package-client.js

```diff
@@ -582,16 +718,26 @@ exports.publishPackage = function (options) {
     // publish a new build.
 
     buildmessage.enterJob("uploading source", function () {
-      uploadTarball(uploadInfo.uploadUrl, sourceBundleResult.sourceTarball);
+      uploadFile(uploadInfo.uploadUrl, sourceBundleResult.sourceTarball);
+    });
+    if (buildmessage.jobHasMessages())
+      return;
+
+    var hashes = {
+      tarballHash: sourceBundleResult.tarballHash,
+      treeHash: sourceBundleResult.treeHash,
+      readmeHash: readmeInfo.hash
+    };
+
+    buildmessage.enterJob("uploading docs", function () {
+      uploadFile(uploadInfo.readmeUrl, readmeInfo.path);
```

Comment:
Oh, here's a potential issue.  Both uploadInfo.uploadUrl and uploadInfo.readmeUrl are given to us as signed URLs that expire in 2 minutes.  So if you're on a slow connection and uploading the source takes two minutes, you might be unable to upload the docs.  (I don't know btw if the expires date is checked at beginning or end of PUT.)

Could be fixed by parallelizing, or for that matter just putting the docs which are probably smaller first.